core/trace/src/main/scala-2/org/typelevel/otel4s/trace/TracerMacro.scala

```diff
@@ -54,7 +52,7 @@ private[otel4s] trait TracerMacro[F[_]] {
     * @param attributes
     *   the set of attributes to associate with the span
     */
-  def span(name: String, attributes: Attribute[_]*): Resource[F, Span[F]] =
+  def span(name: String, attributes: Attribute[_]*): SpanBuilder[F] =
```

Comment:
This could return SpanBuilder or a new SpanOps.  The idea is to have resource-like methods, most importantly `use` and `surround`.

Reply:
I would prefer `SpanOps`. That way we can split responsibilities. 
Also, the implementation of `SpanBuilderImpl` should be much cleaner.

Reply:
Would `SpanBuilder` extend `SpanOps`, or introduce an intermediate call, like fs2's `.compile`?

Reply:
I guess an intermediate call would be better.

We can mimic the existing implementation:
```scala
trait Tracer[F[_]] {
  def span(name: String, ...): SpanOps[F] // basically a shortcut for spanBuilder(...).build
  def spanBuilder(name: String): SpanBuilder[F]
}

trait SpanBuilder[F[_]] {
  def build: SpanOps[F]
}
```